### PR TITLE
MAJOR CHANGE: Restructure repository

### PR DESCRIPTION
**Pull request by Ramimashkouk**
* Make bot/run/* and bot/build/* endpoints work
* Add run/connect endpoint to open websocket

### CODE REVIEW

File: backend/df_designer/app/__init__.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+                *cmd_to_run.split(),
+                stdout=asyncio.subprocess.PIPE,
+                stderr=asyncio.subprocess.PIPE,
+                stdin=asyncio.subprocess.PIPE,
+            )
+            self.pid = self.process.pid
+
+    def check_status(self):
+        if self.process is None:
+            return "Process has not been started."
```
Comment (RLKRo) [inline backend/df_designer/app/__init__.py:25]: Maybe return an Enum instance instead? (and obtain `returncode` separately wherewhere needed)

File: backend/df_designer/app/__init__.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        elif self.process.returncode is None:
+            return "Running"
+        else:
+            return f"Exited with return code: {self.process.returncode}. A non-zero return code indicates an error."
+
+    def stop(self):
+        if self.process is None:  # Check if a process has been started
+            raise RuntimeError("Cannot stop a process '{self.pid}' that has not started yet.")
```
Comment (RLKRo) [inline backend/df_designer/app/__init__.py:33]: A couple of strings here seem to be missing an `f` modifier.

File: backend/df_designer/app/cli.py [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+logger = get_logger(__name__)
+
+cli = typer.Typer()
+
+@cli.command("build_bot")
+def build_bot(
+    project_dir: str = settings.WORK_DIRECTORY,
+    preset: str = "success"
+):
+    presets_build_path = os.path.join(project_dir, "df_designer", "presets", "build.json")
```
Comment (RLKRo) [inline backend/df_designer/app/cli.py:19]: For path handling I prefer using [`pathlib`](https://docs.python.org/3/library/pathlib.html) -- it is more high-level.

File: backend/df_designer/app/__init__.py (hunk 2 above)
Comment (Ramimashkouk) [reply backend/df_designer/app/__init__.py:33]: Fixed in later commits

File: backend/df_designer/app/cli.py (hunk 3 above)
Comment (Ramimashkouk) [reply backend/df_designer/app/cli.py:19]: Resolved in ec931cb9050be8593f990553851cb985985c7171

**Pull request merged**